static/js/components/TruncatedText.js

```diff
@@ -29,6 +29,31 @@ export default function TruncatedText(props: Props) {
     showExpansionControls
   } = props
   const [expanded, setExpanded] = useState(false)
+  const [hasOverflow, setHasOverflow] = useState(false)
+  const [, forceReRender] = useState(0)
+  const dotRef = useRef(null)
+
+  useLayoutEffect(() => {
+    if (dotRef.current) {
+      // the .container on the Dotdotdot holds the ref for the div it uses to
+      // wrap the text
+      const { container } = dotRef.current
+
+      if (container.scrollHeight === 0 && container.offsetHeight === 0) {
+        // the div which holds the text has been added to the DOM but doesn't have
+        // a calculated height yet. in this case we need to use a setState call
+        // to force the component to re-render so we can get the values we need
+        // to figure out when the div has overflow
+        forceReRender(val => val + 1)
```

Comment:
I think this loop is being called endlessly if the resource's short description is "", for example course MITx+12.340x ("Outdated - Global Warming Science") causes the following exception and the page goes blank:

<img width="444" alt="Screen Shot 2019-12-16 at 12 01 42 PM" src="https://user-images.githubusercontent.com/187676/70927578-3a171680-1ffd-11ea-9370-5ec4cab7133e.png">

Reply:
ah, that is not good! will fix haha